db/db_iter.cc

```diff
@@ -709,6 +732,12 @@ bool DBIter::FindValueForCurrentKey() {
   size_t num_skipped = 0;
   while (iter_->Valid() && ikey.sequence <= sequence_ &&
          user_comparator_->Equal(ikey.user_key, saved_key_.GetKey())) {
+    if (TooManyInternalKeysSkipped()) {
+      return false;
+    } else {
+      num_internal_keys_skipped_++;
+    }
+
```

Comment:
I think this part should be on line 789. The first time we enter this loop we don't know whether the key will be  skipped.